client/my-sites/site-settings/settings-section-header.tsx

```diff
@@ -1,40 +1,38 @@
 import { Button } from '@automattic/components';
 import { useTranslate } from 'i18n-calypso';
-import PropTypes from 'prop-types';
+import { FC, PropsWithChildren, ReactNode } from 'react';
 import SectionHeader from 'calypso/components/section-header';
 
-const SettingsSectionHeader = ( {
+interface SettingsSectionHeaderProps extends PropsWithChildren {
```

Comment:
Instead of PropsWithChildren, let's add children inline with a type of `ReactNode`, and make it optional. It's simpler and how it's usually done these days.

```suggestion
interface SettingsSectionHeaderProps {
    children?: ReactNode;
```

Reply:
We are using `PropsWithChildren ` in many places:

fbhepr%2Sfrnepu%3Scebwrpg%3Qpnylcfb%26shyy%3QCebcfJvguPuvyqera%26qrsf%3Q%26ersf%3Q%26cngu%3Q%26uvfg%3Q%26glcr%3Q%26keq%3Q%26aa%3Q1-og

Seems like a matter of taste at the moment. Maybe it's worth to introduce a general rule for it and apply eslint rule or sth to help folks be consistent with it?

Reply:
Yeah that could be a strategy for a larger refactor.

PropsWithChildren (along with FC) were common practice a few years ago, that's probably why it's used in many places, I think it's completely fine not to use it anymore though - components are simply functions and props are objects so the types are straightforward.

That said I don't intend to block the PR on this, of course, it's a very minor detail so I'm fine with whatever you choose :)

Reply:
I think `PropsWithChildren` exist mainly because the `children` prop used to be included automatically in the `FC` type, but then in React 18 this was changed and you need to declare `children` explicitly. `PropsWithChildren` is a neat helper that's useful mainly for codemods. You don't have to use it, but then you need to remember the correct type for `children`: `ReactNode`, optional, etc.

However, one thing that's not very good is using `MyProps extends PropsWithChildren`. It's supposed to be `PropsWithChildren<MyProps>`.

Reply:
Removed `PropsWithChildren` and `FC` usage.

I agree the component looks much simpler now but I would love those types to be banned by tslint or sth to avoid confusion and improve the consistency of our codebase since folks are still using it. Example: https://github.com/Automattic/wp-calypso/pull/91482

Reply:
Yes, they could be added to the `@typescript-eslint/ban-types` rule in the ESLint config.

Here's a Create React App PR from 2019 that removes `FC` from their default templates, and contains a discussion of all the pros and cons: https://github.com/facebook/create-react-app/pull/8177

Reply:
Bookmarking that! Thanks Jarda, very interesting.

Reply:
Thank you, I found it on StackOverflow 🙂